tests/i18n/tests.py

```diff
@@ -1104,45 +1104,42 @@ def test_parse_spec_http_header(self):
-        self.assertEqual([('da', 1.0), ('en', 0.5), ('en-gb', 0.25)], p('da, en-gb;q=0.25, en;q=0.5'))
-        self.assertEqual([('en-au-xx', 1.0)], p('en-au-xx'))
-        self.assertEqual(
-            [('de', 1.0), ('en-au', 0.75), ('en-us', 0.5), ('en', 0.25), ('es', 0.125), ('fa', 0.125)],
-            p('de,en-au;q=0.75,en-us;q=0.5,en;q=0.25,es;q=0.125,fa;q=0.125')
-        )
-        self.assertEqual([('*', 1.0)], p('*'))
-        self.assertEqual([('de', 0.0)], p('de;q=0.'))
-        self.assertEqual([('en', 1.0), ('*', 0.5)], p('en; q=1.0, * ; q=0.5'))
-        self.assertEqual([('en', 1.0)], p('en; q=1,'))
-        self.assertEqual([], p(''))
-
-        # Bad headers; should always return [].
-        self.assertEqual([], p('en-gb;q=1.0000'))
-        self.assertEqual([], p('en;q=0.1234'))
-        self.assertEqual([], p('en;q=.2'))
-        self.assertEqual([], p('abcdefghi-au'))
-        self.assertEqual([], p('**'))
-        self.assertEqual([], p('en,,gb'))
-        self.assertEqual([], p('en-au;q=0.1.0'))
-        self.assertEqual(
-            [],
-            p('XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXZ,en')
-        )
-        self.assertEqual([], p('da, en-gb;q=0.8, en;q=0.7,#'))
-        self.assertEqual([], p('de;q=2.0'))
-        self.assertEqual([], p('de;q=0.a'))
-        self.assertEqual([], p('12-345'))
-        self.assertEqual([], p(''))
-        self.assertEqual([], p('en;q=1e0'))
+        tests = [
+            # Good headers
+            ([('de', 1.0)], 'de'),
+            ([('en-au', 1.0)], 'en-AU'),
+            ([('es-419', 1.0)], 'es-419'),
+            ([('*', 1.0)], '*;q=1.00'),
+            ([('en-au', 0.123)], 'en-AU;q=0.123'),
+            ([('en-au', 0.5)], 'en-au;q=0.5'),
+            ([('en-au', 1.0)], 'en-au;q=1.0'),
+            ([('da', 1.0), ('en', 0.5), ('en-gb', 0.25)], 'da, en-gb;q=0.25, en;q=0.5'),
+            ([('en-au-xx', 1.0)], 'en-au-xx'),
+            ([('de', 1.0), ('en-au', 0.75), ('en-us', 0.5), ('en', 0.25), ('es', 0.125), ('fa', 0.125)],
+             'de,en-au;q=0.75,en-us;q=0.5,en;q=0.25,es;q=0.125,fa;q=0.125'),
+            ([('*', 1.0)], '*'),
+            ([('de', 0.0)], 'de;q=0.'),
+            ([('en', 1.0), ('*', 0.5)], 'en; q=1.0, * ; q=0.5'),
+            # Bad headers; should always return [].
+            ([], 'en-gb;q=1.0000'),
+            ([], 'en;q=0.1234'),
+            ([], 'en;q=.2'),
+            ([], 'abcdefghi-au'),
+            ([], '**'),
+            ([], 'en,,gb'),
+            ([], 'en-au;q=0.1.0'),
+            ([],
+             'XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXZ,en'),
+            ([], 'da, en-gb;q=0.8, en;q=0.7,#'),
+            ([], 'de;q=2.0'),
+            ([], 'de;q=0.a'),
+            ([], '12-345'),
+            ([], ''),
+            ([], 'en;q=1e0')
+        ]
+        for expected, value in tests:
```

Comment:
Could you switch this around so it's `value, expected` -- that's consistent with other tests.